src/components/Autocomplete/components/ComboBox/ComboBox.tsx

```diff
@@ -304,18 +306,20 @@ export default class ComboBox extends React.PureComponent<Props, State> {
     this.handlePopoverOpen;
   };
 
-  private handleEnter = () => {
-    const {selectedOption} = this.state;
-
-    if (this.state.popoverActive && selectedOption) {
-      if (isOption(selectedOption)) {
-        this.handleSelection(selectedOption.value);
-      } else {
-        selectedOption.onAction && selectedOption.onAction();
+  private handleEnter = (event: KeyboardEvent) => {
+    if (event.keyCode === Key.Enter) {
```

Comment:
What do you think about having an early return if `if (event.keyCode !== Key.Enter) return;`so we don't have the entire function in a condition?

Reply:
Good point. I feel early returns is more along the lines of our patterns 👍